src/RecoObjects/Jet.cpp

```diff
@@ -242,12 +242,12 @@ const ParticlePointer Jet::smear_jet(const ParticlePointer jet, const ParticlePo
 		}
 	}
 	//use raw scaleFactors from above to calculate the final factors to apply
-	double matchedGeneratedJetpt = gen_jet->pt();
-	double jetPt = jet->pt();
+	double gen_pt = gen_jet->pt();
+	double reco_pt = jet->pt();
 
-	double deltaPt = ( scaleFactor - 1 ) * (jetPt - matchedGeneratedJetpt);
+	double new_pt = std::max(0.0, gen_pt + ( scaleFactor * (reco_pt - gen_pt)));
 
-	double ptScale = std::max(0.0, ( (jetPt + deltaPt)/jetPt ));
+	double ptScale = new_pt/reco_pt;
```

Comment:
it is identical functionality just rewritten to be closer to what is given on the JER twiki.